### ISSUE TITLE
`kairos-agent upgrade` fusses about `mount: /run/initramfs/isoscan: mount point is busy.`

### ISSUE THREAD

**Issue by jbalonso** [kairos-io/kairos#255]
**TL; DR:**
When attempting an upgrade a VM installed from `kairos-opensuse-v1.1.0-k3sv1.24.6+k3s1.iso`:
```
kairos@kairos-belted-steward:~> sudo kairos-agent upgrade v1.1.3-k3sv1.24.6-k3s1
<--snip-->
INFO[2022-10-20T19:53:06Z] Upgrade completed                            
E1020 19:53:07.188629    8078 mount_linux.go:195] Mount failed: exit status 32
Mounting command: systemd-run
Mounting arguments: --description=Kubernetes transient mount for /run/initramfs/isoscan --scope -- mount -t auto -o remount,ro /dev/sda4 /run/initramfs/isoscan
Output: Running scope as unit: run-r1c51f8442c9a4eacabdedb66567b4cfb.scope
mount: /run/initramfs/isoscan: mount point is busy.

ERRO[2022-10-20T19:53:07Z] Failed mounting device /dev/sda4 with label COS_STATE 
exit status 1
```
Rebooting the node manually (e.g. `exec sudo reboot`) seems to show a clean upgrade, but I feel like this should be smoother.

**Kairos version:**
```
NAME="kairos-opensuse"
VERSION="v1.1.2-k3sv1.24.6+k3s1"
ID="kairos"
ID_LIKE="kairos-opensuse"
VERSION_ID="v1.1.2-k3sv1.24.6+k3s1"
PRETTY_NAME="kairos-opensuse v1.1.2-k3sv1.24.6+k3s1"
ANSI_COLOR="0;32"
BUG_REPORT_URL="https://github.com/kairos-io/kairos/issues/new/choose"
HOME_URL="https://github.com/kairos-io/provider-kairos"
IMAGE_REPO="quay.io/kairos/kairos-opensuse"
IMAGE_LABEL="latest"
GITHUB_REPO="kairos-io/provider-kairos"
VARIANT="core"
FLAVOR="opensuse"
```

**CPU architecture, OS, and Version:**
```
Linux kairos-belted-steward 5.14.21-150400.24.21-default #1 SMP PREEMPT_DYNAMIC Wed Sep 7 06:51:18 UTC 2022 (974d0aa) x86_64 x86_64 x86_64 GNU/Linux
```

**Describe the bug**
When running `kairos-agent upgrade v1.1.3-k3sv1.24.6-k3s1`, the upgrade seems to "succeed", but the process errors out rather than exiting cleanly. I unconfidently suspect that the process should conclude with a node reboot.

I note that these log messages appear in #249 without remark.

**To Reproduce**
Run `kairos-agent upgrade v1.1.3-k3sv1.24.6-k3s1` on a host installed from `kairos-opensuse-v1.1.0-k3sv1.24.6+k3s1.iso`. The reproduction steps for #249 might still produce the same result.

**Expected behavior**
A clean exit.

**Logs**
```
kairos@kairos-belted-steward:~> sudo kairos-agent upgrade v1.1.3-k3sv1.24.6-k3s1
INFO[2022-10-20T19:50:49Z] Starting elemental version v0.0.1            
WARN[2022-10-20T19:50:49Z] failed reading installation state: open /run/initramfs/cos-state/state.yaml: no such file or directory 
INFO[2022-10-20T19:50:49Z] Upgrade called                               
INFO[2022-10-20T19:50:49Z] Applying 'before-upgrade' hook               
INFO[2022-10-20T19:50:49Z] Running before-upgrade hook                  
INFO[2022-10-20T19:50:49Z] deploying image quay.io/kairos/kairos-opensuse:v1.1.3-k3sv1.24.6-k3s1 to /run/initramfs/isoscan/cOS/transition.img 
INFO[2022-10-20T19:50:49Z] Creating file system image /run/initramfs/isoscan/cOS/transition.img 
INFO[2022-10-20T19:50:50Z] Copying quay.io/kairos/kairos-opensuse:v1.1.3-k3sv1.24.6-k3s1 source... 
INFO[0001] Unpacking a container image: quay.io/kairos/kairos-opensuse:v1.1.3-k3sv1.24.6-k3s1 
INFO[0001] Pulling an image from remote repository      
INFO[0128] Pulled: sha256:93f8ab6da984f74c1e0d66183e125735125275dc473583baa2d8b52345215520 quay.io/kairos/kairos-opensuse:v1.1.3-k3sv1.24.6-k3s1 
INFO[0128] Size: 1.295GiB                               
INFO[2022-10-20T19:52:57Z] Finished copying quay.io/kairos/kairos-opensuse:v1.1.3-k3sv1.24.6-k3s1 into /run/cos/transition 
INFO[2022-10-20T19:52:59Z] Applying 'after-upgrade-chroot' hook         
INFO[2022-10-20T19:53:00Z] Running after-upgrade-chroot hook            
INFO[2022-10-20T19:53:00Z] rebranding                                   
INFO[2022-10-20T19:53:00Z] Setting default grub entry to Kairos         
INFO[2022-10-20T19:53:00Z] Applying 'after-upgrade' hook                
INFO[2022-10-20T19:53:00Z] Running after-upgrade hook                   
INFO[2022-10-20T19:53:06Z] Backing up current active image              
INFO[2022-10-20T19:53:06Z] Moving /run/initramfs/isoscan/cOS/active.img to /run/initramfs/isoscan/cOS/passive.img 
INFO[2022-10-20T19:53:06Z] Finished moving /run/initramfs/isoscan/cOS/active.img to /run/initramfs/isoscan/cOS/passive.img 
INFO[2022-10-20T19:53:06Z] Moving /run/initramfs/isoscan/cOS/transition.img to /run/initramfs/isoscan/cOS/active.img 
INFO[2022-10-20T19:53:06Z] Finished moving /run/initramfs/isoscan/cOS/transition.img to /run/initramfs/isoscan/cOS/active.img 
INFO[2022-10-20T19:53:06Z] Upgrade completed                            
E1020 19:53:07.188629    8078 mount_linux.go:195] Mount failed: exit status 32
Mounting command: systemd-run
Mounting arguments: --description=Kubernetes transient mount for /run/initramfs/isoscan --scope -- mount -t auto -o remount,ro /dev/sda4 /run/initramfs/isoscan
Output: Running scope as unit: run-r1c51f8442c9a4eacabdedb66567b4cfb.scope
mount: /run/initramfs/isoscan: mount point is busy.

ERRO[2022-10-20T19:53:07Z] Failed mounting device /dev/sda4 with label COS_STATE 
exit status 1
kairos@kairos-belted-steward:~> exec sudo reboot
```

**Comment by jbalonso** [comment]
I may have identified a further issue. After rebooting, `/etc/os-version` and the k8s node object still report I'm running v1.1.2 and not v1.1.3. As such, it _looks_ like the upgrade didn't go through.

I could have sworn I was able to upgrade from v1.1.0 to v1.1.2 this way, though.

**Comment by jbalonso** [comment]
> I may have identified a further issue. After rebooting, `/etc/os-version` and the k8s node object still report I'm running v1.1.2 and not v1.1.3. As such, it _looks_ like the upgrade didn't go through.
> 
> I could have sworn I was able to upgrade from v1.1.0 to v1.1.2 this way, though.

Sometime when I wasn't looking, the node started reporting that it's running v1.1.3, so I likely made a mistake somewhere and the upgrade is fine after a reboot.

**Comment by mudler** [comment]
awesome, thanks for confirming @jbalonso ! closing the issue for now
